File: src/main/scala/universalteam/flatbedrock/FlatBedrock.scala
package universalteam.flatbedrock

import cpw.mods.fml.common.Mod
import cpw.mods.fml.common.event.{FMLPostInitializationEvent, FMLInitializationEvent, FMLPreInitializationEvent}

@Mod(modid = FlatBedrock.MOD_ID, name = FlatBedrock.MOD_NAME, version = FlatBedrock.MOD_VERSION, modLanguage = "scala", dependencies = "required-after:Forge;after:UniversalCore")
object FlatBedrock
{
    val MOD_ID = "FlatBedrockx"
    val MOD_NAME = "FlatBedrock"
    val MOD_VERSION = "@VERSION@"

    @Mod.EventHandler
    def preInit(event: FMLPreInitializationEvent)
    {
        Proxy.preInit
    }

    @Mod.EventHandler
    def init(event: FMLInitializationEvent)
    {
        Proxy.init
    }

    @Mod.EventHandler
    def postInit(event: FMLPostInitializationEvent)
    {
        Proxy.postInit
    }
}

File: src/main/java/universalteam/flatbedrock/FlatBedrock.java
package universalteam.flatbedrock;

import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLInterModComms;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import universalteam.flatbedrock.handler.IMCHandler;
import universalteam.flatbedrock.lib.Reference;
import universalteam.flatbedrock.proxies.CommonProxy;

@Mod(modid = Reference.MOD_ID, name = Reference.MOD_NAME, version = Reference.MOD_VERSION, dependencies = "after:UniversalCore")
public class FlatBedrock
{
	@Mod.Instance(Reference.MOD_ID)
	public static FlatBedrock instance;

	@SidedProxy(clientSide = "universalteam.flatbedrock.proxies.ClientProxy", serverSide = "universalteam.flatbedrock.proxies.CommonProxy")
	public static CommonProxy proxy;

	@Mod.EventHandler
	public void preInit(FMLPreInitializationEvent event)
	{
		proxy.preInit();
	}

	@Mod.EventHandler
	public void init(FMLInitializationEvent event)
	{
		proxy.init();
	}

	@Mod.EventHandler
	public void postInit(FMLPostInitializationEvent event)
	{
		proxy.postInit();
	}

	@Mod.EventHandler
	public void handleIMCMessages(FMLInterModComms.IMCEvent event)
	{
		IMCHandler.processMessages(event.getMessages());
	}
}
